src/ServiceControl.Monitoring/Http/Diagrams/DiagramApiModule.cs

```diff
@@ -107,8 +147,21 @@ static MonitoredEndpoint[] GetMonitoredEndpoints(EndpointRegistry endpointRegist
                 {
                     Name = endpoint.Key,
                     EndpointInstanceIds = endpoint.Value.Select(i => i.InstanceId).ToArray(),
-                    IsStale = endpoint.Value.Any(i => activityTracker.IsStale(i))
+                    IsStale = endpoint.Value.Any(activityTracker.IsStale)
                 }).ToArray();
         }
+
+        static MonitoredEndpointMessageType[] GetMonitoredMessageTypes(IEnumerable<IProvideBreakdownBy<EndpointMessageType>> stores, HistoryPeriod period, string endpointName)
+        {
+            var messageTypes = stores.SelectMany(s => s.GetIntervals(period, DateTime.UtcNow))
```

Comment:
This looks really heavy. Any tests for 100 endpoints with 100 message types each?

Reply:
see https://github.com/Particular/ServiceControl.Monitoring/pull/54#discussion_r136049553